services/platform/platform.go

```diff
@@ -111,16 +115,53 @@ func (s *Service) Connect(ctx context.Context, req *platformpb.ConnectRequest) (
 	})
 	if err != nil {
 		s.l.Errorf("Failed to insert SSO details: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 
 	if err := s.UpdateSupervisordConfigurations(ctx); err != nil {
 		s.l.Errorf("Failed to update configuration of grafana after connecting PMM to Portal: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 	return &platformpb.ConnectResponse{}, nil
 }
 
+// Disconnect disconnects a PMM server from the organization created on Percona Portal.
+func (s *Service) Disconnect(ctx context.Context, req *platformpb.DisconnectRequest) (*platformpb.DisconnectResponse, error) {
+	ssoDetails, err := models.GetPerconaSSODetails(ctx, s.db.Querier)
+	if err == nil {
+		return nil, status.Error(codes.Aborted, "PMM server is not connected to Portal")
+	}
+
+	settings, err := models.GetSettings(s.db)
+	if err != nil {
+		s.l.Errorf("Failed to fetch PMM server ID and address: %s", err)
+		return nil, status.Error(codes.Internal, internalServerError)
+	}
+
+	nCtx, cancel := context.WithTimeout(ctx, platformAPITimeout)
+	defer cancel()
+
+	err = s.disconnect(nCtx, &disconnectPMMParams{
+		PMMServerID: settings.PMMServerID,
+		AccessToken: ssoDetails.AccessToken.AccessToken,
+	})
+	if err != nil {
+		return nil, err // this is already a status error
+	}
+
+	err = models.DeletePerconaSSODetails(s.db.Querier)
```

Comment:
What do you think about reversing the order? If the deletion fails here, we can't reconnect.
If we delete the sso and then disconnect. When disconnect fails, we can rollback SSO details deleting.

Reply:
> @oksana-grishchenko @jprukner Ok I will reverse order, do we want do this also for connect? Because right now it has same order like disconnect.

Seems like `connect` doesn't have any restore actions. In case of failure it only returns an error and log it, so I guess the order is less matter in that case. 
I just would like to highlight that in the ideal world the consistent state should be restored and in that case the order matters. I don't know if this restore process after failed `disconnect` (as well as `connect`) is in the scope of the current task.